frontend/src/components/homepage/PreviewModal.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState } from "react";
+import { gql, useQuery } from "@apollo/client";
+import {
+  Badge,
+  Box,
+  Button,
+  Flex,
+  Heading,
+  Link,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  ModalCloseButton,
+  Text,
+} from "@chakra-ui/react";
+import convertLanguageTitleCase from "../../utils/LanguageUtils";
+
+export type PreviewModalProps = {
+  storyId: number;
+  title: string;
+  youtubeLink: string;
+  level: number;
+  language: string;
+  previewBook: () => void;
+  preview: boolean;
+  primaryBtnText: string;
+  primaryBtnOnClick: () => any;
+};
+
+const PreviewModal = ({
+  storyId,
+  title,
+  youtubeLink,
+  level,
+  language,
+  previewBook,
+  preview,
+  primaryBtnText,
+  primaryBtnOnClick,
+}: PreviewModalProps) => {
+  const [content, setContent] = useState<string[]>([]);
+
+  const GET_STORY_CONTENTS = (id: number) =>
+    gql`
+      query {
+        storyById(
+          id: ${id}
+        ) {
+          contents {
+            id
+            lineIndex
+            content
+          }
+        }
+      }
+    `;
+
+  useQuery(GET_STORY_CONTENTS(storyId), {
+    fetchPolicy: "cache-and-network",
+    onCompleted: (data) => {
+      const storyContentsArray = data.storyById.contents.map((story: any) => {
+        return story.content;
+      });
+      setContent(storyContentsArray);
+    },
+  });
+
+  return (
+    <Modal
+      isOpen={preview}
+      onClose={previewBook}
+      motionPreset="slideInBottom"
+      size="xl"
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>
+          <Box width="95%">
+            <Box width="70%">
+              <Heading as="h3" size="md">
+                {title}
+              </Heading>
+            </Box>
+            <Box width="75%" padding="10px 0px 10px 0px" float="left">
+              <Badge>{`Level ${level}`}</Badge>
+              <Badge>{`${convertLanguageTitleCase(language)}`}</Badge>
+            </Box>
+            <Box width="25%" float="right">
+              <Button colorScheme="blue" mr={3} onClick={primaryBtnOnClick()}>
+                {primaryBtnText}
+              </Button>
+            </Box>
+          </Box>
```

Comment:
![image](https://user-images.githubusercontent.com/15113249/128605380-caa24b90-e6e0-4375-ac0e-3a7f61d8a26c.png)
the button seems to be aligned with the title (right now it's aligned with nothing), but the design doesnt include the x button. Going to start a slack convo about this